routes/index.js

```diff
@@ -63,23 +67,116 @@ router.get('/central-programs', async (req, res, next) => {
+                                      (SELECT position_id, MAX(assignment_id) as max_assignment
+                                      FROM staffing
+                                       WHERE year = ${year}
+                                      GROUP BY position_id) m
+                                    LEFT JOIN staffing st ON (m.position_id = st.position_id AND m.max_assignment = st.assignment_id)
+                                    LEFT JOIN bargaining_units bu on st.bargaining_unit_id = bu.bargaining_unit
+                                    WHERE
+                                    st.site_code >= 900
+                                    AND st.year = ${year}
+                                    GROUP BY st.site_code, bu.abbreviation, bu.description
+                                    ORDER BY st.site_code ASC`
 
     try {
-        getData(query, res, processor)
+        var programs = await pgPool.query(centralProgramsQuery)
+        programs = programs.rows
+        var staffRoles
+        var rolesGroupedByProgram = {}
+
+        if(includeStaffRoles) {
+            var allStaffRoles = await pgPool.query(staffRolesQuery)
+            staffRoles = allStaffRoles.rows
+            rolesGroupedByProgram = staffRoles.reduce((r,row) => {
+                // Bit of a hack for now
+                // If there is a better role name in the database, send that instead of janky OUSD name
+                if(row.role_display_description) {
+                  row.role_description = row.role_display_description
+                }
+                delete row.role_display_description
+
+                var code = row.site_code
+                delete row.site_code
+
+                r[code] = r[code] || []
+                r[code].push(row)
+                return r
+            }, Object.create(null))
+
+            programs = programs.map(program => {
+                if(program.code in rolesGroupedByProgram) {
+                  program.staff_roles = rolesGroupedByProgram[program.code]
+                } else {
+                  program.staff_roles = []
+                }
+                return program
+            })
+
+        }
+
+        if(includeStaffBargainingUnits) {
+            var allStaffBargainingUnits = await pgPool.query(staffBargainingUnitsQuery)
+            staffBargainingUnits = allStaffBargainingUnits.rows
+            staffBargainingUnitsGroupedByProgram = staffBargainingUnits.reduce((r,row) => {
+                var code = row.site_code
+                delete row.site_code
+
+                r[code] = r[code] || []
+                r[code].push(row)
+                return r
+            }, Object.create(null))
+
+            programs = programs.map(program => {
+                if(program.code in staffBargainingUnitsGroupedByProgram) {
+                  program.staff_bargaining_units = staffBargainingUnitsGroupedByProgram[program.code]
+                } else {
+                  program.staff_bargaining_units = []
```

Comment:
If we don't include an empty property when there is no data, Gatsby throws an error when creating the nodes in the GraphQL db.